packages/contracts-bedrock/test/setup/Upgrade.s.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+// Testing
+import { VmSafe } from "forge-std/Vm.sol";
+import { console2 as console } from "forge-std/console2.sol";
+import { stdJson } from "forge-std/StdJson.sol";
+
+// Scripts
+import { Deployer } from "scripts/deploy/Deployer.sol";
+import { DeploySuperchainInput, DeploySuperchain, DeploySuperchainOutput } from "scripts/deploy/DeploySuperchain.s.sol";
+
+// Libraries
+import { Constants } from "src/libraries/Constants.sol";
+import { GameTypes } from "src/dispute/lib/Types.sol";
+
+// Interfaces
+import { IFaultDisputeGame } from "interfaces/dispute/IFaultDisputeGame.sol";
+import { IDisputeGame } from "interfaces/dispute/IDisputeGame.sol";
+import { IDisputeGameFactory } from "interfaces/dispute/IDisputeGameFactory.sol";
+import { IAddressManager } from "interfaces/legacy/IAddressManager.sol";
+
+/// @title Upgrade
+/// @notice This script is called by Setup.sol as a preparation step for the foundry test suite, and is run as an
+///         alternative to Deploy.s.sol, when `UPGRADE_TEST=true` is set in the env.
+///         Like Deploy.s.sol this script saves the system addresses to disk so that they can be read into memory later
+///         on, however rather than deploying new contracts from the local source code, it simply reads the addresses
+///         from the superchain-registry.
+///         Therefore this script can only be run against a fork of a production network which is listed in the
+///         superchain-registry.
+///         This contract must not have constructor logic because it is set into state using `etch`.
+contract Upgrade is Deployer {
+    using stdJson for string;
+
+    /// @notice Reads a standard chains addresses from the superchain-registry and saves them to disk.
+    function run() public {
+        string memory superchainBasePath = "./lib/superchain-registry/superchain/configs/";
```

Comment:
Could add a CI job to make sure superchain registry commit is up to date

Reply:
I thought about this more since we spoke, and I'm not sure we should break the monorepo CI everytime a commit is made to the superchain registry. That would cause issues in the merge queue for one thing.